### DISCUSSION TITLE
Plausible automated and unattended installation support (create user and web site by script)

### DISCUSSION THREAD

**Discussion by stephane-klein** [plausible/analytics#4547] [Self-Hosted Support]
Hello,

I would like to install and configure Plausible automatically via a script.

Does Plausible support automated and unattended installation?

I'd like to be able to create by script :

- A new user
- Add a website

Do you know if this feature is already available?

If not, what do you think about creating an issue to implement this feature?

Best regards,
Stéphane

**Comment by ruslandoga** [comment]
👋 @stephane-klein 

There is no easy way to do it right now. But you might be able to write a script still, since Plausible supports RPC, e.g.:

```console
$ docker compose exec plausible bin/plausible rpc 'IO.puts("hello")'
hello
```

**Reply by ruslandoga** [reply]
Some relevant functions / modules:
- https://github.com/plausible/analytics/blob/master/lib/plausible/sites.ex#L178-L206
- https://github.com/plausible/analytics/blob/master/lib/plausible_web/live/register_form.ex

Creating a user would be something like
```elixir
Plausible.Repo.insert!(Plausible.Auth.User.new(params))
```
You can also combine the two repo calls in a single translation.
```elixir
Plausible.Repo.transaction(fn ->
  user = Plausible.Repo.insert!(Plausible.Auth.User.new(user_params))
  {:ok, site} = Plausible.Sites.create_site(user, site_params)
  IO.inspect(user, label: "user")
  IO.inspect(site, label: "site")
end)
```
